Remove vision code

DJ-Laser opened the pull request:
## Description

This PR is best categorized by the following descriptors:

- [x] Bugfix
- [x] Refactor
- [ ] New Feature
- [ ] Testing
- [ ] Cleanup/Documentation

### Context

No vision :P

Closes #(issue)

### What Has Changed?
removed mention of vision code and made the classes affected throw errors on instantiation

## Checklists

### Testing

- [ ] This PR has been tested/verified during a meeting with no issues found.
- [ ] This PR has been tested/verified during driver practice with no issues found.
- [ ] This PR has been tested/verified at competition with no issues found.

I built and simulated it to to errors :)

### Formatting

- [ ] This PR passes auto-formatting checks.
- [ ] The code introduced in this PR is self-documenting through reasonable variable and function names.

---
- [ ] This PR is ready to merge. Sufficient reasoning has been given for any
points left unchecked above.

TigerBotsCSP committed:
Remove non java vision code

TigerBotsCSP committed:
Remove mention of left and right cameras

TigerBotsCSP committed:
Extract VisionDevice to abstract class

DJ-Laser committed:
Remove limelight mentions

DJ-Laser committed:
throw if citrus vision device is created

hamster1147 commented on src/main/java/com/team1678/frc2024/Constants.java:
```diff
@@ -542,45 +542,7 @@ public static final class ClimberConstants {
-								"{\"camera_matrix\":[[910.3756558875847,0,809.2765926238984],[0,909.8129438903156,644.2713243574459],[0,0,1]],\"distortion_coefficients\":[0.06236712235474046,-0.062294270427656145,0.004664145480488657,-0.0006911909097633055,-0.00762026244976393],\"tag_size\":0.163,\"camera_settings\":{\"width\":1600,\"height\":1200,\"fps\":50}}")));
-	}
-
+	/*
```
Should just remove these since they won't be relevant to us even when we add vision.

Commit:
Use Java UnsupportedOperationException instead of outside library Exception

Commit:
Remove unused vision-related imports

Commit:
Disable heading zeroing when there are no vision measurements
Otherwise, the code would be setting the gyro to NaN, breaking the gyro code

Commit:
Clean up heading average code
Call a method with the heading measure, instead of adding the number to a MovingAverage exposed by the class
This helps make the code more resuable and hides implementation details of the VisionDeviceManager

Commit:
Add PhotonVision vendordep

Commit:
Move citrus Vision constants to citrus vision device class

Commit:
Add PhotonVisionDevice
This class mirrors the functions of CitrusVisionDevice, but is impemented in photonvision instead of polaris.

Commit:
Add logging and stdev calculations based on distance

hamster1147 committed:
Merge branch 'main' into vision-rip-devin

hamster1147 committed:
Merge branch 'removeLimelight' into vision-rip-devin-no-limelight

hamster1147 commented on src/main/java/com/team1678/frc2024/subsystems/vision/VisionDevice.java:
```diff
@@ -1,243 +1,7 @@
-		List<VisionFrame> frames = new ArrayList<VisionFrame>();
-		boolean is_connected;
-	}
+public abstract class VisionDevice extends Subsystem {
+	abstract boolean isConnected(); 
 }
```
Can just be an interface at this point.

hamster1147 replied:
Didn't notice that it still extends Subsystem. I guess that makes it a little funky trying to make it an interface.

hamster1147 committed:
Apply spotless

hamster1147 committed:
Remove commented polaris code and fix access modifier
